tests/DCPS/DynamicData/publisher.cpp

```diff
@@ -41,11 +56,18 @@ int ACE_TMAIN(int argc, ACE_TCHAR *argv[])
 
   HelloWorld::MessageTypeSupport_var type_support = new HelloWorld::MessageTypeSupportImpl();
   CORBA::String_var type_name = type_support->get_type_name();
-
-  DDS::DynamicType_var dt = type_support->get_type();
-  DDS::TypeSupport_var dts = new DDS::DynamicTypeSupport(dt);
-  if (check_rc(dts->register_type(participant, type_name), "register_type failed")) {
-    return 1;
+  DDS::DynamicType_var dt;
+
+  if (dynamic) {
+    dt = type_support->get_type();
+    DDS::TypeSupport_var dts = new DDS::DynamicTypeSupport(dt);
+    if (check_rc(dts->register_type(participant, type_name), "register_type (dynamic) failed")) {
+      return 1;
+    }
+  } else {
+    if (check_rc(type_support->register_type(participant, type_name), "register_type (plain) failed")) {
+      return 1;
+    }
   }
```

Comment:
I'm not sure about this one... it's an if/else at the top level for two somewhat parallel concepts.

Reply:
I'm ok with keeping it. The changes I suggested save a pair of brackets because the top level else doesn't do anything else other than `check_rc`, but it's maybe harder to read.